zebra/dplane_fpm_nl.c

```diff
@@ -1260,7 +1265,10 @@ static int fpm_process_queue(struct thread *t)
 		if (ctx == NULL)
 			break;
 
-		fpm_nl_enqueue(fnc, ctx);
+		if (fpm_nl_enqueue(fnc, ctx) == -1)
```

Comment:
Are you going to do this everywhere else? Why is this special?

Also this situation should never happen, because we check for space at the beginning of the loop:

```c
		/* No space available yet. */
		if (STREAM_WRITEABLE(fnc->obuf) < NL_PKT_BUF_SIZE)
			break;
```

I understand it showed up in the coverity scan result, but I think we should ignore it.

Reply:
the return is being tested in all of the other places, so this is making it consistent (and quieting coverity)

Reply:
@mjstapp 

> the return is being tested in all of the other places, so this is making it consistent (and quieting coverity)

Sure, but the other places are not doing:

```c
			atomic_fetch_add_explicit(
				&fnc->counters.dplane_contexts_failed, 1,
				memory_order_relaxed);
```

(also making it inconsistent again)

Maybe it would be better to just add a `zlog_warn`, `zlog_err` or `assert` to make it quiet in this case.

Reply:
duh - quite right. maybe the management of the counter could go into `fpm_nl_enqueue` then?